cpp/src/arrow/python/python_to_arrow.cc

```diff
@@ -191,10 +191,11 @@ struct ValueConverter<Date64Type> {
 
 template <>
 struct ValueConverter<Time32Type> {
-  static inline Result<int32_t> FromPython(PyObject* obj, TimeUnit::type unit) {
+  static inline Result<int32_t> FromPython(PyObject* obj, TimeUnit::type unit,
+                                           bool /*ignore_timezone*/) {
```

Comment:
This is a bit ugly, could we pass an options struct instead?

Reply:
My thought here was this parameter will only be need until the next release, so it would be better to add it here and then remove it without the need to introduce options.  We can introduce options though if you feel strongly.